### ISSUE TITLE
Implement new Process API

### ISSUE THREAD

**Issue by c4spar** [c4spar/deno-dzx#7]
This is a draft for the new `Process` class.

```ts
declare class Process
  implements
    Promise<ProcessOutput>,
    AsyncIterableIterator<string>,
    Deno.Reader,
    Deno.Writer {
  [Symbol.toStringTag]: string;
  [Symbol.asyncIterator](): AsyncIterator<string>;
  readonly rid: number;
  readonly pid: number;
  readonly stdin: Deno.Writer;
  readonly stdout: Deno.Reader & Promise<string> & { copy(dest: Deno.Writer | Process): Process };
  readonly stderr: Deno.Reader & Promise<string> & { copy(dest: Deno.Writer | Process): Process };
  readonly status: Promise<Deno.ProcessStatus>;
  readonly statusCode: Promise<number>;
  readonly noThrow: this;
  kill(signo: number): void;
  read(p: Uint8Array): Promise<number | null>;
  write(p: Uint8Array): Promise<number>;
  copy(p: Uint8Array): Promise<number>;
  then<T = ProcessOutput, S = never>(
    resolve?:
      | ((processOutput: ProcessOutput) => T | PromiseLike<T>)
      | undefined
      | null,
    reject?:
      | ((error: unknown) => S | PromiseLike<S>)
      | undefined
      | null,
  ): Promise<T | S>;
  catch<T = never>(
    reject?:
      | ((error: unknown) => T | PromiseLike<T>)
      | undefined
      | null,
  ): Promise<ProcessOutput | T>;
  finally(onFinally?: (() => void) | null): Promise<ProcessOutput>;
}
```

**Comment by sigmaSd** [comment]
I think the only missing thing compared to zx, is `pipe` function which could be a very nice addition.

**Comment by c4spar** [comment]
New api is implemented in #39
